e2e/pull/pull.go

```diff
@@ -30,79 +30,84 @@ type ctx struct {
-	library         string // use specific library, XXX(mem): not tested yet
-	force           bool   // pass --force
-	createDst       bool   // create destination file before pull
-	unauthenticated bool   // pass --allow-unauthenticated
-	setImagePath    bool   // pass destination path
-	setPullDir      bool   // pass --dir
-	expectSuccess   bool   // singularity should exit with code 0
-}{
+type testStruct struct {
+	desc             string // case description
+	srcURI           string // source URI for image
+	library          string // use specific library, XXX(mem): not tested yet
+	force            bool   // pass --force
+	createDst        bool   // create destination file before pull
+	unauthenticated  bool   // pass --allow-unauthenticated
+	setImagePath     bool   // pass destination path
+	setPullDir       bool   // pass --dir
+	expectedExitCode int
+	pullDir          string
+	imagePath        string
+	expectedImage    string
+}
+
+var tests = []testStruct{
 	{
-		desc:          "non existent image",
-		srcURI:        "library://sylabs/tests/does_not_exist:0",
-		expectSuccess: false,
+		desc:             "non existent image",
+		srcURI:           "library://sylabs/tests/does_not_exist:0",
+		expectedExitCode: 255,
 	},
 
 	// --allow-unauthenticated tests
 	{
-		desc:            "unsigned image allow unauthenticated",
-		srcURI:          "library://sylabs/tests/unsigned:1.0.0",
-		unauthenticated: true,
-		expectSuccess:   true,
+		desc:             "unsigned image allow unauthenticated",
+		srcURI:           "library://sylabs/tests/unsigned:1.0.0",
+		unauthenticated:  true,
+		expectedExitCode: 0,
 	},
 
 	// --force tests
 	{
-		desc:            "force existing file",
-		srcURI:          "library://alpine:3.8",
-		force:           true,
-		createDst:       true,
-		unauthenticated: true,
-		expectSuccess:   true,
+		desc:             "force existing file",
+		srcURI:           "library://alpine:3.8",
+		force:            true,
+		createDst:        true,
+		unauthenticated:  true,
+		expectedExitCode: 0,
 	},
 	{
-		desc:            "force non-existing file",
-		srcURI:          "library://alpine:3.8",
-		force:           true,
-		createDst:       false,
-		unauthenticated: true,
-		expectSuccess:   true,
+		desc:             "force non-existing file",
+		srcURI:           "library://alpine:3.8",
+		force:            true,
+		createDst:        false,
+		unauthenticated:  true,
+		expectedExitCode: 0,
 	},
 	{
 		// --force should not have an effect on --allow-unauthenticated=false
-		desc:            "unsigned image force require authenticated",
-		srcURI:          "library://sylabs/tests/unsigned:1.0.0",
-		force:           true,
-		unauthenticated: false,
-		expectSuccess:   false,
+		desc:             "unsigned image force require authenticated",
+		srcURI:           "library://sylabs/tests/unsigned:1.0.0",
+		force:            true,
+		unauthenticated:  false,
+		expectedExitCode: 10,
```

Comment:
Is `10` intended here? Why is the return value `10` if so?

Reply:
Based on my understanding of the code an exit code of 10 is related to an error while pulling unsigned libraries (error occurring while exporting keys also trigger an exit code equal to 10 based on `key_export.go`): https://github.com/sylabs/singularity/blob/a23832994ed18d47eb2ec651a37100cc4043fbc4/cmd/internal/cli/pull.go#L218

I can add a comment if you want